spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/VaultEnvironmentRepositoryFactory.java

```diff
@@ -25,22 +25,36 @@
 
 /**
  * @author Dylan Roberts
+ * @author Scott Frederick
  */
 public class VaultEnvironmentRepositoryFactory implements
 		EnvironmentRepositoryFactory<VaultEnvironmentRepository, VaultEnvironmentProperties> {
 
-	private ObjectProvider<HttpServletRequest> request;
+	private final ObjectProvider<HttpServletRequest> request;
 
-	private EnvironmentWatch watch;
+	private final EnvironmentWatch watch;
 
-	private Optional<VaultRestTemplateFactory> vaultRestTemplateFactory;
+	private final Optional<VaultRestTemplateFactory> vaultRestTemplateFactory;
+
+	private final ConfigTokenProvider tokenProvider;
 
 	public VaultEnvironmentRepositoryFactory(ObjectProvider<HttpServletRequest> request,
 			EnvironmentWatch watch,
 			Optional<VaultRestTemplateFactory> vaultRestTemplateFactory) {
 		this.request = request;
 		this.watch = watch;
 		this.vaultRestTemplateFactory = vaultRestTemplateFactory;
+		this.tokenProvider = null;
```

Comment:
Can we just use the http token provider by default here?

Reply:
I was trying to avoid duplicating that defaulting logic in both `VaultEnvironmentRepository` and `VaultEnvironmentRepositoryFactory`, and it has to be in the former. Not a huge amount of duplication so I'm happy to do it either way.

Reply:
Done